test/tools/Modules/HelpersHostCS/HelpersHostCS.psm1

```diff
@@ -145,18 +146,63 @@ namespace TestHost
 
         public override PSCredential PromptForCredential(string caption, string message, string userName, string targetName)
         {
-            Streams.Prompt.Add("Credential:" + caption + ":" + message);
-            SecureString ss = ReadLineAsSecureString();
-            string userNameToUse = string.IsNullOrEmpty(userName) ? UserNameForCredential : userName;
-            return new PSCredential(userNameToUse, ss);
+            return PromptForCredential(caption,
+                                        message,
+                                        userName,
+                                        reenterPassword: false,
+                                        targetName,
+                                        PSCredentialTypes.Default,
+                                        PSCredentialUIOptions.Default);
         }
 
         public override PSCredential PromptForCredential(string caption, string message, string userName, string targetName, PSCredentialTypes allowedCredentialTypes, PSCredentialUIOptions options)
+        {
+            return PromptForCredential(caption,
+                                         message,
+                                         userName,
+                                         reenterPassword: false,
+                                         targetName,
+                                         allowedCredentialTypes,
+                                         options);
+        }
+
+        public override PSCredential PromptForCredential(string caption, string message, string userName, bool reenterPassword, string targetName, PSCredentialTypes allowedCredentialTypes, PSCredentialUIOptions options)
         {
             Streams.Prompt.Add("Credential:" + caption + ":" + message);
-            SecureString ss = ReadLineAsSecureString();
+            SecureString password = ReadLineAsSecureString();
+            if(reenterPassword)
+            {
+                SecureString confirmPassword = ReadLineAsSecureString();
```

Comment:
The test host emulates user IO for the cmdlet. That allows to test the cmdlet but not real host - real host can be tested only manually.

We need to check three scenario:
1. ReenterPassword is false
2. ReenterPassword is true and first password confirmation is right
3. ReenterPassword is true,  first password confirmation is wrong,  second password confirmation is right.

As I said, real host can be tested only manually so the test host can address only:
1. ReenterPassword is false
Test output is `Streams.Prompt.Add("Credential:" + caption + ":" + message);`
2. ReenterPassword is true
Test output could be `Streams.Prompt.Add("Credential@" + caption + "@" + message);`

Reply:
@iSazonov I have made changes to the test case and helper module. Please review and provide your feedback.